snap/types.go

```diff
@@ -200,6 +200,25 @@ type Component struct {
 	Type        ComponentType `yaml:"type"`
 	Summary     string        `yaml:"summary"`
 	Description string        `yaml:"description"`
+
+	// HookNames contains a mapping of hook names to plugs, exactly as it was
+	// defined in the snap.yaml file for this component. The plugs that are
+	// listed in HookNames will not be the complete list of plugs that are bound
+	// to each hook. For example, an unscoped plug will be bound to all hooks in
+	// all components.
+	HookNames map[string]ComponentHookPlugs `yaml:"hooks"`
```

Comment:
the name of this is a bit strange, maybe HookDefs or HookDecls?  also we will need to support everything that goes into hookYaml I think

Reply:
I know that @alfonsosanchezbeato made you undo that, but is a bit unclear to me why this is cleaner than having a componentYaml given that we need to use hookYaml anyway?

Reply:
given the direction of travel I would say to go with componentYaml,  it fits with the rest and given the increased complexity seem clearer overall